.github/workflows/gradle.yml

```diff
@@ -0,0 +1,29 @@
+name: Java CI
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Grant execute permission for gradlew
+      run: chmod +x gradlew
+    - name: Test with Gradle
+      run: ./gradlew test
+    - name: Build with Gradle
+      run: ./gradlew build
+    - name: Deploy to AWS
+      uses: appleboy/lambda-action@master
+      with:
```

Comment:
how about we replace the `with: ` part with this:

```
        env:
          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
        run: aws lambda update-function-code --function-name mparticle_iterable_firehose_test --zip-file fileb://iterable-extension/build/distributions/iterable-extension.zip
```

This way, we just use the `awscli` that comes with the Github action container and avoid using external git repos

Reply:
The `https://github.com/stcalica/update-lambda/blob/master/index.js` that you suggested does the same thing in JS 

```
const core = require('@actions/core');
const github = require('@actions/github');
const fs = require('fs');
const AWS = require('aws-sdk');

try {
  const functionName = core.getInput('function-name');
  const package = core.getInput('package');
  const AWS_SECRET_KEY = core.getInput('AWS_SECRET_KEY');
  const AWS_SECRET_ID = core.getInput('AWS_SECRET_ID');
  const AWS_REGION = core.getInput('AWS_REGION');

  console.log(`Deploying ${functionName} from ${package}.`);

  var zipBuffer = fs.readFileSync(`./${package}`);
  core.debug('ZIP file put into memory buffer.');

  const lambda = new AWS.Lambda({
      apiVersion: '2015-03-31',
      region: AWS_REGION,
      secretAccessKey: AWS_SECRET_KEY,
      accessKeyId: AWS_SECRET_ID,
      maxRetries: 3,
      sslEnabled: true,
      logger: console,
  });

  const params = {
    FunctionName: functionName,
    Publish: true,
    ZipFile: zipBuffer,
  };

  lambda.updateFunctionCode(params, err => {
      if (err) {
          console.error(err);
          core.setFailed(err)
      }
  });

} catch (error) {
  core.setFailed(error.message);
}
```

Essentially, we are just calling `update-function-code` in either case, which can be done with a single `aws-cli` call

Reply:
We can also create a `Deploy to prd` step with the conditional `if: github.ref == 'master'` if you'd like

Reply:
Updated, it works!